tdrs-backend/tdpservice/users/management/commands/populate_stts.py

```diff
@@ -0,0 +1,80 @@
+"""`populate_stts` command."""
+
+import csv
+from pathlib import Path
+
+from django.core.management import BaseCommand
+
+from ...models import Region, STT
+
+
+DATA_DIR = BASE_DIR = Path(__file__).resolve().parent / "data"
+
+
+def _populate_regions():
+    with open(DATA_DIR / "regions.csv") as csvfile:
+        reader = csv.DictReader(csvfile)
+        for row in reader:
+            Region.objects.get_or_create(id=row["Id"])
+        Region.objects.get_or_create(id=1000)
+
+
+def _get_states():
+    with open(DATA_DIR / "states.csv") as csvfile:
+        reader = csv.DictReader(csvfile)
+        return [
+            STT(
+                code=row["Code"],
+                name=row["Name"],
+                region_id=row["Region"],
+                type=STT.STTType.STATE,
+            )
+            for row in reader
+        ]
+
+
+def _get_territories():
+    with open(DATA_DIR / "territories.csv") as csvfile:
+        reader = csv.DictReader(csvfile)
+        return [
+            STT(
+                code=row["Code"],
+                name=row["Name"],
+                region_id=row["Region"],
+                type=STT.STTType.TERRITORY,
+            )
+            for row in reader
+        ]
+
+
+def _populate_tribes():
+    with open(DATA_DIR / "tribes.csv") as csvfile:
+        reader = csv.DictReader(csvfile)
+        stts = [
+            STT(
+                name=row["Name"],
+                region_id=row["Region"],
+                state=STT.objects.get(code=row["Code"]),
+                type=STT.STTType.TRIBE,
+            )
+            for row in reader
+        ]
+        STT.objects.bulk_create(stts, ignore_conflicts=True)
+
+
+class Command(BaseCommand):
+    """Command class."""
+
+    help = "Populate regions, states, territories, and tribes."
+
+    def handle(self, *args, **options):
+        """Populate the various regions, states, territories, and tribes."""
+        _populate_regions()
+        stts = _get_states()
+        stts.extend(_get_territories())
+        STT.objects.bulk_create(stts, ignore_conflicts=True)
```

Comment:
If this command were run against a live DB that already had users that had associated STT, how would that be handled?

Reply:
ATM this would just create duplicates, which we obviously don't want. We're discussing strategies for how to handle this most efficiently. We could check to see if data already exists, and then not create it if it does, but then we would have to loop and check each one individually. Not too difficult, but not so efficient either. Was also thinking of adding a unique constraint on the STT name and then catching the error if it tries to duplicate a record. Probably leaning toward the latter. What do you think @lauraGgit ?

Reply:
Actually, the ignore_conflicts parameter already nicely ignores any duplicates, so this wouldn't affect any existing records or cause any other problems. If a new STT were added it would add the new record, but that wouldn't affect existing records.